### DISCUSSION TITLE
Beancount 3 vs Beancount 2

### DISCUSSION THREAD

**Discussion by slifty** [beancount/beancount#783] [Q&A] [answered]
I've been lurking on the mailing list and I get the sense that there is some major momentum around Beancount v3.

The README says `This is unstable and you want to use version 2 below` -- is that still the case, or should I try to update my pip files to pull v3?

I noticed that the published v3 release [is from 2022](https://pypi.org/project/beancount/#history), so I was also wondering what needs to happen for the project to feel comfortable starting to release v3 builds more formally.

**Comment by blais** [comment, marked as answer]
You know when you get immersed a lot on a project and there are a lot of things you start to see you could do better but they don't necessarily suggest a coherent redo of the entire thing?  And then one day it clicks, you just see how you could make everything better by rearranging things in a particular way?  That's what happened between v1 and v2. 

Then I'd been using v2 for a long time and at some point the same thing happened for v3, I saw how it could be rearranged and simplified--further!-- to make something better. And I wrote down all the ideas, and I was going to do a rewrite. I wrote a new parser, a new build, I got started. And then life got in the way: a new demanding job, another child, and not enough time to put in to play, in fact, not enough time to do anything. Rewriting as v3 and implementing all those things remains something I'd love to do, I just don't have enough time anymore. I wouldn't call it "abandoned", the dream is alive, I use v2 daily and a v3 version is something I think about every single time I run bean-check. I have no idea when I'll be able to do this.

I would encourage someone to take those ideas and write something new from scratch. Of course, someone else will have their own ideas and do something else.  The problem with new projects is that it's easy to get started, but in order to make them mature and robust it takes ~10x more effort and precious few projects get to that stage (github is largely a cemetary of incomplete things). v2 is the result of that effort.
